DATA_FORMAT.md

```diff
@@ -0,0 +1,330 @@
+
+### 2. Times
+
+The times JSON is similar to the tracks JSON. It also contains a `data` array that maps from track IDs to the frame number that they appear on.
+
+```
+--times.json--
+{
+    "data": [
+        <frame number for track id 0>,
+        <frame number for track id 1>,
+        <frame number for track id 2>,
+        ...
+    ]
+}
+```
+
+### 3. Frames
+
+_Example frame:_
+![](./documentation/frame_0.png)
+_Each unique color in this frame is a different track ID._
+
+**Frames** are image textures that store the track IDs for each time step in the time series. Each pixel in the image can encode a single track ID in its RGB value (`= R + G*256 + B*256*256`). Note that there is currently no way to indicate overlaps in the data-- your script will need to flatten 3D data and segmentations.
+
+There should be one frame for every time step in the time series, and they must all be listed in order in the **manifest** file.
+
+### 4. Features
+
+Datasets can contain any number of features, which are a numeric value assigned to each track ID in the dataset. Each feature file usually corresponds to a single column of data.
+
+Features must also provide a `min` and `max` range property.
+
+```
+--feature1.json--
+{
+    "data": [
+        <feature value for track id 0>,
+        <feature value for track id 1>,
+        <feature value for track id 2>,
+        ...
+    ],
+    "min": <min value for all features>,
+    "max": <max value for all features>
+}
+```
+
+### 5. Centroids
+
+The centroids file defines the center of each track ID in the dataset. For each index `i`, the coordinates are `(x: data[2i], y: data[2i + 1])`.
+Coordinates are defined in pixels in the frame, where the upper left corner of the frame is (0, 0).
+
+```
+--centroids.json--
+{
+    "data": [
+        <x coordinate for track id 0>,
+        <y coordinate for track id 0>,
+        <x coordinate for track id 1>,
+        <y coordinate for track id 1>,
+        ...
+    ]
+}
+```
+
+### 6. Bounds
+
+The bounds file defines the rectangular boundary occupied by each track ID. For each track ID `i`, the minimum bounding box coordinates (upper left corner) are given by
+`(x: data[4i], y: data[4i + 1])`, and the maximum bounding box coordinates (lower right corner) are given by `(x: data[4i + 2], y: data[4i + 3])`.
+
+Again, coordinates are defined in pixels in the image frame, where the upper left corner is (0, 0).
+
+```
+--bounds.json--
+{
+    "data": [
+        <upper left x for track id 0>,
+        <upper left y for track id 0>,
+        <lower right x for track id 0>,
+        <lower right y for track id 0>,
+        <upper left x for track id 1>,
+        <upper left y for track id 1>,
+        ...
+    ]
+}
+```
+
+### 7. Outliers
+
+The outliers file stores marks whether a given track ID should be marked as an outlier using an array of booleans (`true`/`false`). Indices that are `true` indicate outlier values, and are given a unique color in Nucmorph-Colorizer.
```

Comment:
```suggestion
The outliers file stores whether a given track ID should be marked as an outlier using an array of booleans (`true`/`false`). Indices that are `true` indicate outlier values, and are given a unique color in Nucmorph-Colorizer.
```

Reply:
I wonder if this should be considered optional data for some datasets.  It seemed silly to generate a file with all false's in it.  Also, this is likely to be sparse data and could possibly be stored as such.  We can wait to hear from users on that...

Reply:
Thanks for catching this!

It actually turns out that quite a few of the files *are* optional, and I went back and marked them in the documentation. Centroids, bounds, and outliers are all optional. 
(Technically, so are tracks and times, but I think the app will stop working as expected without them.)

Reply:
We can't do "show path" without centroids.  So I'd only say it's optional if that is handled gracefully.  And yes, it seems to me that tracks and times SHOULD be required!  And at least one feature (until we support color-by-track-id)!

Reply:
The show path option currently just won't do anything without centroid data. In the future I could disable the checkbox in the UI and add a tooltip indicating why it's disabled...